src/commands/project/deploy/report.ts

```diff
@@ -51,23 +57,77 @@ export default class DeployMetadataReport extends SfCommand<DeployResultJson> {
       summary: messages.getMessage('flags.results-dir.summary'),
       helpGroup: testFlags,
     }),
+    // we want to allow undefined for a simple check deploy status
+    // eslint-disable-next-line sf-plugin/flag-min-max-default
+    wait: Flags.duration({
+      char: 'w',
+      summary: deployMessages.getMessage('flags.wait.summary'),
+      description: deployMessages.getMessage('flags.wait.description'),
+      unit: 'minutes',
+      helpValue: '<minutes>',
+      min: 1,
+    }),
   };
 
   public async run(): Promise<DeployResultJson> {
     const [{ flags }, cache] = await Promise.all([this.parse(DeployMetadataReport), DeployCache.create()]);
-    const jobId = cache.resolveLatest(flags['use-most-recent'], flags['job-id']);
+    const jobId = cache.resolveLatest(flags['use-most-recent'], flags['job-id'], false);
+
+    const deployOpts = cache.get(jobId) ?? ({} as DeployOptions & { isMdapi: boolean });
+    const waitDuration = flags['wait'];
+    const org = flags['target-org'] ?? (await Org.create({ aliasOrUsername: deployOpts['target-org'] }));
 
-    const deployOpts = cache.get(jobId);
-    const org = await Org.create({ aliasOrUsername: deployOpts['target-org'] });
-    const [deployStatus, componentSet] = await Promise.all([
-      // we'll use whatever the org supports since we can't specify the org
+    // if we're using mdapi we won't have a component set
+    let componentSet = new ComponentSet();
+    if (!deployOpts.isMdapi) {
+      if (!cache.get(jobId)) {
+        // If the cache file isn't there, use the project package directories for the CompSet
+        try {
+          this.project = await SfProject.resolve();
+          const sourcepath = this.project.getUniquePackageDirectories().map((pDir) => pDir.fullPath);
+          componentSet = await buildComponentSet({ 'source-dir': sourcepath, wait: waitDuration });
+        } catch (err) {
+          // ignore the error. this was just to get improved command output.
+        }
+      } else {
+        componentSet = await buildComponentSet({ ...deployOpts, wait: waitDuration });
+      }
+    }
+    const mdapiDeploy = new MetadataApiDeploy({
+      // setting an API version here won't matter since we're just checking deploy status
       // eslint-disable-next-line sf-plugin/get-connection-with-version
```

Comment:
that assumes that the mdapi output never changes?  What if they did need to make a breaking change, or added something to the reponse that required a minimum version otherwise it wouldn't appear, but the users wanted it (or didn't)?

It's probably true, until it isn't.

Reply:
I'll update the comment to be more accurate.  When you do `Org.create(...)` it will use `ConfigAggregator` to set the api version.